libimplantisomd5.c

```diff
@@ -31,240 +31,139 @@
+        printf("fragmd5 = %s\n", fragmentsums);
+        printf("frags = %ld\n", FRAGMENT_COUNT);
     }
-    /*    memcpy(new_appdata, orig_appdata, 512); */
-    memset(new_appdata, ' ', 512);
+    memset(appdata, ' ', APPDATA_SIZE);
 
-    loc = 0;
-    loc = writeAppData(new_appdata, "ISO MD5SUM = ", loc);
-    loc = writeAppData(new_appdata, md5str, loc);
-    loc = writeAppData(new_appdata, ";", loc);
+    size_t loc = 0;
+    loc = writeAppData(appdata, "ISO MD5SUM = ", loc);
+    loc = writeAppData(appdata, hashsum, loc);
+    loc = writeAppData(appdata, ";", loc);
 
-    buf = malloc(512 * sizeof(unsigned char));
-    snprintf((char *)buf, 512, "SKIPSECTORS = %d", SKIPSECTORS);
+    char *appdata_buffer;
+    appdata_buffer = aligned_alloc(pagesize, APPDATA_SIZE * sizeof(*appdata_buffer));
+    snprintf(appdata_buffer, APPDATA_SIZE, "SKIPSECTORS = %lld", SKIPSECTORS);
 
-    loc = writeAppData(new_appdata, (char *)buf, loc);
-    loc = writeAppData(new_appdata, ";", loc);
-    free(buf);
+    loc = writeAppData(appdata, appdata_buffer, loc);
+    loc = writeAppData(appdata, ";", loc);
 
-    if (supported) {
-        if (!quiet)
-            printf("Setting supported flag to 1\n");
-        loc = writeAppData(new_appdata, "RHLISOSTATUS=1", loc);
-    } else {
-        if (!quiet)
-            printf("Setting supported flag to 0\n");
-        loc = writeAppData(new_appdata, "RHLISOSTATUS=0", loc);
-    }
+    if (!quiet)
+        printf("Setting supported flag to %d\n", supported);
+    static const char status[] = "RHLISOSTATUS=%d";
+    char tmp[sizeof(status) / sizeof(*status)];
+    snprintf(tmp, sizeof(status) / sizeof(*status), status, supported);
+    loc = writeAppData(appdata, tmp, loc);
 
-    loc = writeAppData(new_appdata, ";", loc);
+    loc = writeAppData(appdata, ";", loc);
 
-    loc = writeAppData(new_appdata, "FRAGMENT SUMS = ", loc);
-    loc = writeAppData(new_appdata, fragstr, loc);
-    loc = writeAppData(new_appdata, ";", loc);
+    loc = writeAppData(appdata, "FRAGMENT SUMS = ", loc);
+    loc = writeAppData(appdata, fragmentsums, loc);
+    loc = writeAppData(appdata, ";", loc);
 
-    buf = malloc(512 * sizeof(unsigned char));
-    snprintf((char *)buf, 512, "FRAGMENT COUNT = %d", FRAGMENT_COUNT);
-    loc = writeAppData(new_appdata, (char *)buf, loc);
-    loc = writeAppData(new_appdata, ";", loc);
-    free(buf);
+    snprintf(appdata_buffer, APPDATA_SIZE, "FRAGMENT COUNT = %ld", FRAGMENT_COUNT);
+    loc = writeAppData(appdata, appdata_buffer, loc);
+    loc = writeAppData(appdata, ";", loc);
 
-    loc = writeAppData(new_appdata, "THIS IS NOT THE SAME AS RUNNING MD5SUM ON THIS ISO!!", loc);
+    loc = writeAppData(
+        appdata, "THIS IS NOT THE SAME AS RUNNING MD5SUM ON THIS ISO!!", loc);
 
-    i = lseek(isofd, pvd_offset + APPDATA_OFFSET, SEEK_SET);
-    if (i<0)
+    if (lseek(isofd, pvd_offset + APPDATA_OFFSET, SEEK_SET) < 0)
         printf("seek failed\n");
```

Comment:
This should return a error to the caller, unless you can think of any cases where the seek can fail but the subsequent write is ok.